workflow/sync/multi_throttler_test.go

```diff
@@ -0,0 +1,171 @@
+package sync
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	fakewfclientset "github.com/argoproj/argo-workflows/v3/pkg/client/clientset/versioned/fake"
+)
+
+func TestMultiNoParallelismSamePriority(t *testing.T) {
+	throttler := NewMultiThrottler(0, make(map[string]int), 0, func(Key) {})
+
+	throttler.Add("default/c", 0, time.Now().Add(2*time.Hour))
+	throttler.Add("default/b", 0, time.Now().Add(1*time.Hour))
+	throttler.Add("default/a", 0, time.Now())
+
+	assert.True(t, throttler.Admit("default/a"))
+	assert.True(t, throttler.Admit("default/b"))
+	assert.True(t, throttler.Admit("default/c"))
+}
+
+func TestMultiNoParallelismMultipleBuckets(t *testing.T) {
+	throttler := NewMultiThrottler(1, make(map[string]int), 1, func(Key) {})
+	throttler.Add("a/0", 0, time.Now())
+	throttler.Add("a/1", 0, time.Now().Add(-1*time.Second))
+	throttler.Add("b/0", 0, time.Now().Add(-2*time.Second))
+	throttler.Add("b/1", 0, time.Now().Add(-3*time.Second))
+
+	assert.True(t, throttler.Admit("a/0"))
+	assert.False(t, throttler.Admit("a/1"))
+	assert.False(t, throttler.Admit("b/0"))
+	assert.False(t, throttler.Admit("b/1"))
+	throttler.Remove("a/0")
+	assert.True(t, throttler.Admit("b/1"))
+}
+
+func TestMultiWithParallelismLimitAndPriority(t *testing.T) {
+	queuedKey := ""
+	throttler := NewMultiThrottler(2, make(map[string]int), 2, func(key string) { queuedKey = key })
```

Comment:
The original test had no namespace parallelism, so the second 2 here perhaps should be 0 to better mimic that test.